Friendly-plans/app/src/main/java/pg/autyzm/friendly_plans/manager_app/view/child_list/ChildListActivity.java

```diff
@@ -107,12 +127,24 @@ private Long addChild(String firstName, String lastName) {
         }
     }
 
-    private void removeChild(long itemId){
-            childRepository.delete(itemId);
-            setUpViews();
-            toastUserNotifier.displayNotifications(
-                    R.string.child_removed_message,
-                    getApplicationContext());
+    private void removeChild(long itemId) {
+        childRepository.delete(itemId);
+        setUpViews();
+        toastUserNotifier.displayNotifications(
+                R.string.child_removed_message,
+                getApplicationContext());
+    }
+
+    private void setActiveChild() {
+        Child selectedChild = childListAdapter.getChild(selectedChildPosition);
+        childRepository.setIsActive(selectedChild, true);
+
+        List<Child> childList = childRepository.getAll();
```

Comment:
Because we know only one child can be active at the one time, I would consider updating all children with is active false first, **by db query** and then setting one selected to active. At the moment number of children + 1(get all) queries to db are performed, and if you change the code it would be only 2. (Even for 2 children always 3 calls to db) It will be better for performance and the performance in android apps is the key point. Maybe we won't have so many children that it would affect app's performance much, but I think it is always better to think about it at this step then investigating problems later.